ultimate-messenger/src/main/java/ru/progrm_jarvis/ultimatemessenger/format/model/AsmGeneratingTextModelFactory.java

```diff
@@ -0,0 +1,407 @@
+         */
+        GENERIC_CLASS_SIGNATURE
+                = "<T:" + OBJECT_JVM_CLASS_NAME_WITH_DESCRIPTOR + '>' + OBJECT_JVM_CLASS_NAME_WITH_DESCRIPTOR
+                + TEXT_MODEL_JVM_CLASS_NAME_WITH_GENERIC_AND_DESCRIPTOR,
+        /**
+         * Generic signature of {@link TextModel#getText(Object)} method
+         */
+        GET_TEXT_METHOD_JVM_GENERIC_SIGNATURE
+                = '(' + PARENT_GENERIC_JVM_CLASS_NAME_WITH_DESCRIPTOR + ')' + STRING_JVM_CLASS_NAME_WITH_DESCRIPTOR,
+        /**
+         * Signature of {@link Object#toString()} method
+         */
+        TO_STRING_METHOD_SIGNATURE = "()" + STRING_JVM_CLASS_NAME_WITH_DESCRIPTOR,
+        /**
+         * Generic signature of {@link StringBuilder#append(String)} method
+         */
+        APPEND_STRING_METHOD_SIGNATURE
+                = '(' + STRING_JVM_CLASS_NAME_WITH_DESCRIPTOR + ')' + STRING_BUILDER_JVM_CLASS_NAME_WITH_DESCRIPTOR;
+
+        /* *************************************** Precomputed string lengths *************************************** */
+        /**
+         * Length of {@link #TEXT_MODEL_JVM_CLASS_NAME_WITH_DESCRIPTOR}
+         */
+        private final int TEXT_MODEL_JVM_CLASS_NAME_WITH_DESCRIPTOR_LENGTH
+                = TEXT_MODEL_JVM_CLASS_NAME_WITH_DESCRIPTOR.length(),
+        /**
+         * Length of {@link #TEXT_MODEL_JVM_CLASS_NAME_WITH_GENERIC_AND_DESCRIPTOR}
+         */
+        TEXT_MODEL_JVM_CLASS_NAME_WITH_GENERIC_AND_DESCRIPTOR_LENGTH
+                = TEXT_MODEL_JVM_CLASS_NAME_WITH_GENERIC_AND_DESCRIPTOR.length();
+
+        /* *********************************** Constant arrays of JVM type names *********************************** */
+        /**
+         * Array whose only value is {@link #TEXT_MODEL_JVM_CLASS_NAME}.
+         */
+        protected static final String[] TEXT_MODEL_JVM_CLASS_NAME_ARRAY = new String[]{TEXT_MODEL_JVM_CLASS_NAME};
+
+        /* ****************************************** Stored multi-opcodes ****************************************** */
+        /**
+         * Result of {@link Opcodes#ACC_PUBLIC} and {@link Opcodes#ACC_FINAL} flags disjunction
+         */
+        private static final int OPCODES_ACC_PUBLIC_FINAL = ACC_PUBLIC | ACC_FINAL;
+        /**
+         * Result of {@link Opcodes#ACC_PUBLIC}, {@link Opcodes#ACC_FINAL}
+         * and {@link Opcodes#ACC_SUPER} flags disjunction
+         */
+        private static final int OPCODES_ACC_PUBLIC_FINAL_SUPER = OPCODES_ACC_PUBLIC_FINAL | ACC_SUPER;
+
+        //</editor-fold>
+
+        /**
+         * Converts the given class name to the corresponding bytecode class name.
+         *
+         * @param className name of the class at runtime (using dot delimiter)
+         * @return name of the class in bytecode
+         *
+         * @see #toBytecodeClassName(Class)
+         */
+        protected static String toBytecodeClassName(@NonNull final String className) {
```

Comment:
The ASM API already has such a method. See [org.objectweb.asm.Type#getType(java.lang.String)](https://asm.ow2.io/javadoc/org/objectweb/asm/Type.html#getType-java.lang.String-)